File: blog_springBoot/src/main/java/com/synergism/blog/api/userAPI/entity/UserInformation.java
package com.synergism.blog.api.userAPI.entity;

import com.synergism.blog.core.user.entity.User;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

/**
 * 用户信息实体类
 */
@Getter
@Setter
public class UserInformation{
    //头像
    private String icon = "";
    //昵称
    private String name = "";
    //账号
    private String username = "";
    //生日
    private Date birthday = new Date();
    //性别代码
    private Integer sex = 0;
    //个人简介
    private String intro = "";

    /**
     * 空参构造函数
     */
    public UserInformation(){
    }

    /**
     * 构造函数
     * @param icon 头像
     * @param name 昵称
     * @param username 账号
     * @param birthday 生日
     * @param sex 性别代码
     * @param intro 个人简介
     */
    public UserInformation(String icon, String name, String username, Date birthday, Integer sex, String intro) {
        this.icon = icon;
        this.name = name;
        this.username = username;
        this.birthday = birthday;
        this.sex = sex;
        this.intro = intro;
    }

    /**
     * 从用户中获得用户信息
     * @param user 用户
     * @return 用户信息
     */
    public static UserInformation getInstance(User user){
        return new UserInformation(
                user.getIcon(),
                user.getName(),
                user.getUsername(),
                user.getBirthday(),
                user.getSex(),
                user.getIntro()
        );
    }
}

File: blog_vue/src/model/user/UserInformation.ts
export default class UserInformation {
  //头像
  icon: string;
  //昵称
  nickname: string;
  //账号
  username: string;
  //生日
  birthday: Date;
  //性别代码
  sex: number;
  //性别
  sexName: string;
  //个人简介
  intro: string;
  //园龄
  upToNow: string;

  constructor(
    icon?: string,
    nickname?: string,
    username?: string,
    birthday?: Date,
    sex?: number,
    intro?: string,
    upToNow?: string
  ) {
    this.icon = icon || "";
    this.nickname = nickname || "";
    this.username = username || "";
    this.birthday = birthday || new Date();
    this.sex = sex || 0;
    this.sexName = "";
    this.intro = intro || "";
    this.upToNow = upToNow || "";
  }
}
